juju/juju
# [JUJU-2334] Add worker to rotate secret backend tokens

Add a worker and facade to perform secret backend token rotation.
The implementation is based on what was done for rotating secrets, including watcher and worker design.
We add a new collection to track the rotation times for each backend, and the worker watches this collection.
The facade deals with providing the watcher and also updating the rotation time.

## Checklist


## QA steps

To test, I hacked the minimum rotation time to 1 minute.
Add a vault backend with a token rotate time, eg
`juju add-secret-backend myvault vault --config ~/vaultcfg.yaml token-rotate=1m`
Check logs to see that token rotation happens.
